src/bin/MolSim.cpp

```diff
@@ -87,15 +128,31 @@ auto main(int argc, char *argv[]) -> int {
   auto config = config::Config::parse_config(argc, argv);
   LoggerManager::setup_logger(*config);
 
-  auto simulator = setup<2>(config);
+  // Needs to be initialized and get overwritten
+  std::chrono::high_resolution_clock::time_point startTime;
 
-  const auto startTime = std::chrono::high_resolution_clock::now();
+  if (config->dimensions == 2) {
 
-  if (config->output_frequency == 0) {
-    simulator.run<false>();
+    auto simulator = setup<2>(config);
+    startTime = std::chrono::high_resolution_clock::now();
+    if (config->output_frequency == 0) {
+      simulator.run<false>();
+    } else {
+      simulator.run<true>();
+    }
+  } else if (config->dimensions == 3) {
+
+    auto simulator = setup<3>(config);
+    startTime = std::chrono::high_resolution_clock::now();
+    if (config->output_frequency == 0) {
+      simulator.run<false>();
+    } else {
+      simulator.run<true>();
+    }
```

Comment:
A nice way to reduce code duplication in cases like here can be lambda expressions.

```suggestion

    auto runSimulation = [&](auto &simulator) {
      if (config->output_frequency == 0) {
        simulator.run<false>();
      } else {
        simulator.run<true>();
      }
    };
      
    auto simulator = setup<2>(config);
    startTime = std::chrono::high_resolution_clock::now();
    runSimulation(simulator);
  } else if (config->dimensions == 3) {
    auto simulator = setup<3>(config);
    startTime = std::chrono::high_resolution_clock::now();
    runSimulation(simulator);
```